### PR TITLE
feat: `badges` generator

### PR DESCRIPTION
**Pull request by cpreston321**
### 🔗 Linked issue

#3 

### ❓ Type of change

- [ ] 📖 Documentation (updates to the documentation, readme, or JSdoc annotations)
- [ ] 🐞 Bug fix (a non-breaking change that fixes an issue)
- [ ] 👌 Enhancement (improving an existing functionality like performance)
- [x] ✨ New feature (a non-breaking change that adds functionality)
- [ ] 🧹 Chore (updates to the build process or auxiliary tools and libraries)
- [ ] ⚠️ Breaking change (fix or feature that would cause existing functionality to change)

### 📚 Description

Resolves: #3 

- Added support for AUTOMD args with `-` e.g. `version-name="hello"` [Test Here](https://regex101.com/r/k76gXH/1)
- Added custom support for overwrites if needed.

### 📝 Checklist

- [x] I have linked an issue or discussion.
- [x] I have updated the documentation accordingly.

### CODE REVIEW

File: src/generators/badges.ts [hunk 1]
```diff
@@ -0,0 +1,96 @@
...
+        // Make sure the value is a string and exists
+        if (args[`${k}-${p}`] && typeof args[`${k}-${p}`] === "string") {
+          if (p === "enabled") {
+            badges[k][p] = args[`${k}-${p}`] === "true" ? "true" : "false";
+            continue;
+          }
+          badges[k][p] = args[`${k}-${p}`];
+        }
+      }
+    }
```
Comment (cpreston321) [inline src/generators/badges.ts:63]: I could possibly turn this into an object then use `defu` to merge but I wanted to be more granular with arguments.

Comment (uncenter) [reply src/generators/badges.ts:63]: Seems fine to me 👍

File: src/generators/badges.ts [hunk 2]
```diff
@@ -0,0 +1,96 @@
...
+  let str = "";
+
+  for (const key in badges) {
+    const badge = badges[key];
+
+    if (badge.enabled === "false") {
+      continue;
+    }
+
+    str += `[![${badge.name}](${badge.src})](${badge.href})\n`;
```
Comment (uncenter) [inline src/generators/badges.ts:81]: I know it is just a few characters of a template string but we might as well use `omark.image()` and `omark.link()` here right?

**Comment by cpreston321** [comment]
@pi0 the new _parse doesn't support `-` args does it?

**Comment by pi0** [comment]
@cpreston321 seems working on playground

image

**Comment by cpreston321** [comment]
@pi0 I just tested it too 👍🏼 Works like a charm should prob add to test case.

File: README.md [hunk 3]
```diff
@@ -172,6 +174,38 @@ The `pm-x` generator generates commands for running a package through JavaScript
...
+    [![npm downloads](https://img.shields.io/npm/dm/package-name?style=flat&colorA=18181B&colorB=F0DB4F)](https://npmjs.com/package/package-name)
+
+    <!-- /automd -->
+
+#### Arguments
+
+- `name`: The package name (by default tries to read from the `name` field in `package.json`).
+- `style`: The badge style (defaults to `flat`).
+- `colorA`: The left color of the badge (defaults to `18181B`).
+- `colorB`: The right color of the badge (defaults to `F0DB4F`).
```
Comment (uncenter) [inline README.md:200]: Something about `colorA` and `colorB` doesn't sit right with me. What about `color-left` and `color-right`? Is that too complicated? I think it also makes it more self-explanatory.

EDIT: Huh reading through I guess those are the names of the actual image's query params...

EDIT2: Those query params are not documented though, maybe deprecated in favor of the `labelColor` and `color` params? Maybe `label-color` and `bg-color` could be the argument names?

File: README.md [hunk 4]
```diff
@@ -172,6 +174,38 @@ The `pm-x` generator generates commands for running a package through JavaScript
...
+- `downloads-enabled`: Show npm downloads badge (defaults to `true`).
+- `version-enabled`: Show npm version badge (defaults to `true`).
+- `codecov-enabled`: Show codecov badge (defaults to `false`).
+- `bundle-enabled`: Show bundle size badge (defaults to `false`).
+
+> [!NOTE]
+> Each badge `downloads|version|codecov|bundle` can be enabled or disabled using `*-enabled` arguments also can be overridden with custom URLs using `*-src`, `*-href` and badge name `*-name` arguments.
```
Comment (uncenter) [inline README.md:207]: Probably would be good to give an example of this here.

Comment (cpreston321) [reply README.md:207]: Should be good to go 👍🏼

File: README.md (hunk 3 above)
Comment (cpreston321) [reply README.md]: Should be good now 👍🏼

File: README.md [hunk 5]
```diff
@@ -172,6 +174,45 @@ The `pm-x` generator generates commands for running a package through JavaScript
...
+- `labelColor`: The Background color of the left part of the badge (defaults to `18181B`).
+- `downloads-enabled`: Show npm downloads badge (defaults to `true`).
+- `version-enabled`: Show npm version badge (defaults to `true`).
+- `codecov-enabled`: Show codecov badge (defaults to `false`).
+- `bundle-enabled`: Show bundle size badge (defaults to `false`).
+
+> [!NOTE]
+> Each badge `downloads|version|codecov|bundle` can be enabled or disabled using `*-enabled` arguments also can be overridden with custom URLs using `*-src`, `*-href` and badge name `*-name` arguments.
+
+    <!-- automd:badges name=package-name version-name="npm pkg version" color-left=FFF codecov-enabled no-downloads-enabled -->
```
Comment (uncenter) [inline README.md:209]:
```suggestion
    <!-- automd:badges name="package-name" version-name="npm pkg version" color-left="FFF" codecov-enabled no-downloads-enabled -->
```

**Comment by pi0** [comment]
(working on some local refactors. please don't push more commits likely they will be overridden)

File: README.md (hunk 5 above)
Comment (pi0) [reply README.md:209]: without now works too

Comment (uncenter) [reply README.md:209]: Following the style of HTML attributes is cleaner and is what we should be encouraging tbh.

**Comment by cpreston321** [comment]
@pi0 Just saw this :( Sorry. I accidentally pushed read me updates.

File: README.md [hunk 6]
```diff
@@ -172,6 +174,45 @@ The `pm-x` generator generates commands for running a package through JavaScript
...
+    [![npm downloads](https://img.shields.io/npm/dm/package-name?style=flat&colorA=18181B&colorB=F0DB4F)](https://npmjs.com/package/package-name)
+
+    <!-- /automd -->
+
+#### Arguments
+
+- `name`: The package name (by default tries to read from the `name` field in `package.json`).
+- `style`: The badge style (defaults to `flat`).
+- `color`: The Background color of the right part of the badge (defaults to `F0DB4F`).
+- `label-color`: The Background color of the left part of the badge (defaults to `18181B`).
```
Comment (uncenter) [inline README.md:200]: Don't commit this suggestion right away but that should be lowercase.
```suggestion
- `color`: The background color of the right part of the badge (defaults to `F0DB4F`).
- `label-color`: The background color of the left part of the badge (defaults to `18181B`).
```

**Comment by pi0** [comment]
No worries @cpreston321. I have pushed some changes in 94fca6839b1e2b3f4e94ffac000e638b75375f45

In summary:
- We now also infer `github` from pkg too to use it in URLs that need it
- Updated implementation to support badgen (which is faster) and shields
- Defenitions are now more strict as result. We check for `no-npmDownloads` and `no-npmVersion` (I had some reasons for this convention in future) strictly

I hope you are happy with them and ideas welcome!

File: README.md [hunk 7]
```diff
@@ -172,6 +174,39 @@ The `pm-x` generator generates commands for running a package through JavaScript
...
+- `name`: The npm package name. By default tries to infer from `package.json`
+- `github`: Github repository name. By default tries to infer from `package.json`
+- `bundlephobia`: Show [bundle-phobia](https://bundlephobia.com/) badge. (requires `name`)
+- `codecov`: Enable [codecov](https://codecov.io) badge (requires `github`)
+- `no-npmDownloads`: Hide npm downloads badge
+- `no-npmVersion`: Hide npm version badge
+- `provider`: Can be one of the `shields` (for [shields.io](https://shields.io/)) or `badgen` / `badgenClassic` (for [badgen.net](https://badgen.net/)). Default is `badgen`.
+
+> [!TIP]
+> You can use additional `style`, `labelColor` and `color` args for `shields` provider for example: `provider=shidls style=flat-square labelColor=f0db4f color=18181b`
```
Comment (uncenter) [inline README.md:208]:
```suggestion
- `name`: The npm package name. By default tries to infer from `package.json`.
- `github`: Github repository name. By default tries to infer from `package.json`.
- `bundlephobia`: Show [bundle-phobia](https://bundlephobia.com/) badge (requires `name`).
- `codecov`: Enable [codecov](https://codecov.io) badge (requires `github`).
- `no-npmDownloads`: Hide npm downloads badge.
- `no-npmVersion`: Hide npm version badge.
- `provider`: Can be one of `shields` (for [shields.io](https://shields.io/)) or `badgen` / `badgenClassic` (for [badgen.net](https://badgen.net/)). Default is `badgen`.

> [!TIP]
> You can use additional `style`, `labelColor` and `color` args for `shields` provider for example: `provider=shields style=flat-square labelColor=f0db4f color=18181b`.
```

**Pull request merged**